From c5e950e3cef89a79cafd08cc58df978e08b4dae6 Mon Sep 17 00:00:00 2001
From: Rob Woolley <rob.woolley@windriver.com>
Date: Thu, 28 Aug 2025 07:53:47 -0700
Subject: [PATCH] Remove buildpath

Adding pal_statistics_INCLUDE_DIRS causes the absolute path to be
included in export_hardware_interfaceExport.cmake.

Including the CMake target in target_link_lirbaries() instead
automatically adds the include directory at compile time. This
also results in the target being used in the exported cmake file.

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b250e3ecb..c78408ede 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,6 @@ target_compile_features(hardware_interface PUBLIC cxx_std_17)
 target_include_directories(hardware_interface PUBLIC
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include/hardware_interface>
-  ${pal_statistics_INCLUDE_DIRS}
 )
 target_link_libraries(hardware_interface PUBLIC
                       rclcpp::rclcpp
@@ -52,7 +51,8 @@ target_link_libraries(hardware_interface PUBLIC
                       rcpputils::rcpputils
                       tinyxml2
                       ${joint_limits_TARGETS}
-                      ${pal_statistics_LIBRARIES}
+                      ${pal_statistics_TARGETS}
+                      ${pal_statistics_msgs_TARGETS}
                       ${control_msgs_TARGETS}
                       ${lifecycle_msgs_TARGETS}
                       fmt::fmt)
